Source/WebCore/loader/PolicyChecker.cpp

```diff
@@ -168,7 +171,7 @@ void FrameLoader::PolicyChecker::checkNavigationPolicy(ResourceRequest&& request
     loader->setLastCheckedRequest(ResourceRequest(request));
 
     // Only the PDFDocument iframe is allowed to navigate to webkit-pdfjs-viewer URLs
-    bool isInPDFDocumentFrame = m_frame.ownerElement() && m_frame.ownerElement()->document().isPDFDocument();
+    bool isInPDFDocumentFrame = frameOwnerElement && frameOwnerElement->document().isPDFDocument();
```

Comment:
Isn't it possible that frame had navigated away / removed between we store frameOwnerElement and we get here? Shouldn't we check that the frame owner element is still the same?

Reply:
I will add an assertion for good measure. I believe it has to be the same. All the code that runs in between early returns.

Reply:
ok